### ISSUE TITLE
dApp - sd-6 - Failure due to network connection

### ISSUE THREAD

**Issue by glitchminer** [status-im/status-desktop#15598]
# Bug Report

## Description

1. If there is no internet connection during either the URI validation or connection initiation an error is thrown in console but no feedback is provided in the UI. 

```
failed: Error in connection establishment: net::ERR_NAME_NOT_RESOLVED" file=webpack://wallet_connect_integration/./node_modules/@walletconnect/jsonrpc-ws-connection/dist/index.es.js?:10 category=js
```

2. When the connection is restored further attempts at connecting continue to fail without any feedback in UI. Had to restart to progress.

#### Expected behavior

1. User is aware of current state of the connection
2. App can recovers after failure to process connection attempts

### Additional Information

- Status desktop version: Master, dapp flag
- Operating System: Mac

**Comment by stefandunca** [comment]
I upgraded the priority and impact for this issue to blocker/critical because I found out today that, in some cases, the underlying runtime that runs Wallet Connect SDK is not recovering when the internet connection is established. 

I also spent considerable time finding a solution while exhausting the regular way of managing the WalletEngine in QML.

Therefore the alternative left are more complicated and will take more debugging and experimenting time.
